solr/core/src/test/org/apache/solr/cloud/LeaderElectionTest.java

```diff
@@ -82,17 +83,24 @@ public void setUp() throws Exception {
     seqToThread = Collections.synchronizedMap(new HashMap<Integer,Thread>());
   }
   
-  class TestLeaderElectionContext extends ShardLeaderElectionContextBase {
-    private long runLeaderDelay = 0;
+  static class TestLeaderElectionContext extends ShardLeaderElectionContextBase {
+    private final ZkStateReader zkStateReader;
+    private final long runLeaderDelay;
 
     public TestLeaderElectionContext(LeaderElector leaderElector,
         String shardId, String collection, String coreNodeName, ZkNodeProps props,
         ZkStateReader zkStateReader, long runLeaderDelay) {
-      super (leaderElector, shardId, collection, coreNodeName, props, zkStateReader);
+      super(leaderElector, shardId, collection, coreNodeName, props, zkStateReader);
+      this.zkStateReader = zkStateReader;
       this.runLeaderDelay = runLeaderDelay;
     }
 
     @Override
+    public TestLeaderElectionContext copy() {
+      return new TestLeaderElectionContext(leaderElector, shardId, collection, id, leaderProps, zkStateReader, runLeaderDelay);
+    }
```

Comment:
There was nothing relying on the previous superclass behavior. Also, I've now added a test that calls `retryElection`, which requires an actual (non-explosive) implementation of `copy`.